apps/ejabberd/src/mod_vcard.erl

```diff
@@ -539,7 +541,7 @@ prepare_vcard_search_params(User, VHost, VCARD) ->
 
     US = {LUser, VHost},
 
-    if
+    case
```

Comment:
If only the `internal-server-error` is returned in case of `{error, badarg`} returned from this function, your suggestion will be enough.